src/hypothesis/internal/conjecture/utils.py

```diff
@@ -68,7 +68,11 @@ def byte_distribution(random, n):
         else:
             probe = upper - v
         return int_to_bytes(probe, n)
-    probe = int_from_bytes(data.draw_bytes(nbytes, byte_distribution)) & mask
+
+    probe = gap + 1
+    while probe > gap:
+        probe = int_from_bytes(data.draw_bytes(nbytes, byte_distribution)) & mask
```

Comment:
This line is currently too long (81 characters I think). If you run 'make format' and commit the changes it will fix that for you.